### PR TITLE
Latex caption + cross-referencing in bookdown

### PR DESCRIPTION
**Pull request by nielsbock**
This is a fix for  #1610 and #818. This PR adds a caption component to the LaTex outputs and enables cross-referencing in **bookdown**.

Adding a caption with `tab_caption()`
```
mtcars[1:5, ] %>%
  gt() %>% 
  tab_caption(md("This is a caption for **mtcars**")) %>% 
  as_latex() %>% 
  as.character()
```
results in the following Latex code:
```
"\\begin{table}[!t]\n\\caption{This is a caption for \\textbf{mtcars}} \n\\fontsize{12.0pt}{14.4pt}\\selectfont\n\\begin{tabular*}{\\linewidth}{@{\\extracolsep{\\fill}}rrrrrrrrrrr}\n\\toprule\nmpg & cyl & disp & hp & drat & wt & qsec & vs & am & gear & carb \\\\ \n\\midrule\\addlinespace[2.5pt]\n21.0 & 6 & 160 & 110 & 3.90 & 2.620 & 16.46 & 0 & 1 & 4 & 4 \\\\ \n21.0 & 6 & 160 & 110 & 3.90 & 2.875 & 17.02 & 0 & 1 & 4 & 4 \\\\ \n22.8 & 4 & 108 & 93 & 3.85 & 2.320 & 18.61 & 1 & 1 & 4 & 1 \\\\ \n21.4 & 6 & 258 & 110 & 3.08 & 3.215 & 19.44 & 1 & 0 & 3 & 1 \\\\ \n18.7 & 8 & 360 & 175 & 3.15 & 3.440 & 17.02 & 0 & 0 & 3 & 2 \\\\ \n\\bottomrule\n\\end{tabular*}\n\\end{table}\n"
```

I've updated knitr helper functions in `knitr-utils.R` which enables cross-referencing using **bookdown**. This only works in HTML if ```options("htmltools.preserve.raw" = FALSE)```, which was already commented in the code. For the Latex output in Quarto, I've used the function `check_quarto()` to disable captions created with `tab_captions` as it is not compatible with `tbl-cap`. This is not an issue with HTML in quarto; however, `tbl-cap` should still be used for cross-referencing.

### CODE REVIEW

File: tests/testthat/test-l_table_parts.R [hunk 1]
```diff
@@ -1,3 +1,28 @@
+test_that("A gt table contains the expected caption component", {
+
+  # Create a `tbl_latex` object with `gt()`; this table contains a caption
+  tbl_latex <-
+    gt(mtcars_short) %>%
+    tab_caption("test caption")
```
Comment (olivroy) [inline tests/testthat/test-l_table_parts.R:6]:
```suggestion
    tab_caption("test caption") %>%
    tab_options(latex.use_longtable = TRUE)
```

File: tests/testthat/test-l_table_parts.R [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
+
+  # Perform a snapshot test where a LaTeX table
+  # contains only a caption
+  mtcars_short %>%
+    gt() %>%
+    tab_caption("test caption") %>%
+    tab_options(latex.use_longtable = TRUE) %>%
+    as_latex() %>%
+    as.character() %>%
+    expect_snapshot()
```
Comment (olivroy) [inline tests/testthat/test-l_table_parts.R:23]:
```suggestion
  expect_snapshot_latex(tbl_latex)
```

I refactored the tests recently in #1844, so I suggested changes to conform to the new style

**Review by olivroy — COMMENTED** [review]
I am not using bookdown, so can't comment on this, but I made some suggestions in the tests to conform to new test style.

**Comment by nielsbock** [comment]
Thanks @olivroy. I modified the test to match the new style.

**Comment by olivroy** [comment]
Thanks for working on this and addressing the comments! 

@rich-iannone I don't have much objection here. If this can help some bookdown users, we may as well go for it! 

As long as it doesn't break the  Quarto workflow which should still be to use `tbl-cap:` and `label: tbl-*`. i.e. `tab_caption()` is ignored in Quarto output iiuc.

**Comment by nielsbock** [comment]
No problem!

Yes, it shouldn't break Quarto workflow, as it's disabled in Quarto in the same way as the `latex.tbl.pos` argument in `tab_options()`, using the `check_quarto()` function. I've used `gt` with this code in both Quarto and Rmarkdown for academic work and haven't stumbled upon issues yet.

**Review by rich-iannone — APPROVED** [review]
LGTM!

**Pull request merged**